### ISSUE TITLE
When text is input in a text field, hint text should not be announced in talkback mode

### ISSUE THREAD

**Issue by jiahaog** [flutter/flutter#117943]
Internal bug: b/264211793. 

This issue is copied from the above bug, filed by `customer: money`.

## Steps to Reproduce

Hint text is announced with no input:

![5io7qmX9TUKai96](https://user-images.githubusercontent.com/7111741/210475622-be3b9f3c-cb1d-4c6e-9fb2-1fdae727360b.png)

Hint text is announced with input:

![64CemDjyawGhiTT](https://user-images.githubusercontent.com/7111741/210475646-18fa6f7d-b10f-4cb9-974a-865d42672380.png)

**Expected results:** 

When text is input in a text field, hint text is not expected to be announced in talk back mode.

**Actual results:** 

Hint text is announced in talk back mode

<details>
<summary>Code sample</summary>

```dart
Scaffold(
  appBar: AppBar(title: Text('Flutter Demo Home Page')),
  body: TextField(
    decoration: InputDecoration(hintText: 'Send a message...'),
  ),
)
```

</details>

**Comment by hangyujin** [comment]
Related issue: https://github.com/flutter/flutter/issues/114511

We already excluded hint text semantics when text field has input and label text is provided. The reason we didn't exclude the hint text when text field has input and no label text is provided is that we think the hint text is likely the label.

@goderbauer Do you think we should exclude the hint text semantics when there is an input no matter if there is a label text or not?

**Comment by Hixie** [comment]
FWIW I would recommend not trying to "do the right thing" in order to handle broken apps, just do the technically correct thing and let apps fail. Otherwise, what happens is that people start depending on the heuristics and then you quickly get to a place where the real API is the heuristics and they are hard to explain, hard to debug, and hard to use.
